client/sites-dashboard-v2/index.tsx

```diff
@@ -73,6 +84,27 @@ const SitesDashboardV2 = ( {
 		}
 	}, [ dataViewsState.selectedItem ] );
 
+	const setQueryParams = useCallback(
+		( queryParams: SitesDashboardQueryParams ) => {
+			// There is a chance that the URL is not up to date when it mounts, so delay
+			// the updateQueryParams call to avoid it getting the incorrect URL and then
+			// redirecting back to the previous path.
+			if ( window.location.pathname.startsWith( `/${ section?.group }` ) ) {
+				updateQueryParams( queryParams );
+			} else {
+				window.setTimeout( () => updateQueryParams( queryParams ) );
+			}
+		},
```

Comment:
I pulled in this functionality and comment from our previous sites dashboard.

I made this its own function `setQueryParams` instead of using it directly in the `dataViewsState.search` dependent `useEffect` below because I believe we will reuse this in future PRs in other useEffect hooks to preserve state items in the url such as status filter.

Reply:
Actually I don't think this makes as much sense as it did in my previous exploration. I don't see any reason not to use the same useEffect for all items that would update the url params anyways, and we can extract this separately if needed. Im going to push a quick update here.